proposals/p2875.md

```diff
@@ -0,0 +1,559 @@
+
+```
+fn G() -> i32 {
+  // ✅ `my_f` has function type `FType`. This is a direct call to `F`.
+  return my_f(1);
+}
+
+fn Sort[template F:! type, T:! type](v: Vector(T)*, f: F);
+fn Compare(a: i32, b: i32) -> Ordering { return a.(Ordered.Compare)(b); }
+fn SortInts(v: Vector(i32)*) { Sort(v, Compare); }
+```
+
+For the purpose of the
+[orphan rule](/docs/design/generics/details.md#orphan-rule), a function type is
+considered to be declared by the function declaration that introduces the
+function value.
+
+### Bound methods
+
+For each function type corresponding to a method, there is a corresponding
+_bound method type_. When a member access is performed on an object of class
+type to access a method, the result is a _bound method value_ of bound method
+type. A bound method type describes the callee in a method call, and a bound
+method value describes the `self` parameter of the call.
+
+```
+class HasMember {
+  // `HasMember.F` has a stateless function type, with signature
+  // `[self: Self](n: i32) -> i32`.
+  fn F[self: Self](n: i32) -> i32;
+}
+fn F(h1: HasMember, h2: HasMember) -> i32 {
+  // ✅ `h1.F` is a bound method value whose type is a bound method type,
+  // with signature `(n: i32) -> i32`.
+  var hf: auto = h1.F;
+  // ✅ `h1.F` and `h2.F` are of the same bound method type.
+  hf = h2.F;
+  // ✅ Same as `h2.F(4)`.
+  return hf(4);
+}
+```
+
+### Call syntax
+
+Calls take the form `a(b, c, d)` or `a(b, c, d,)`, where:
+
+-   `a` is the callee, which can be a name, a literal, a member access, or some
+    more complex expression enclosed parentheses.
+-   `b`, `c`, `d` are any number of argument expressions. Arguments are
+    separated by commas, and if the argument list is not empty, an optional
+    trailing comma is permitted but not required after the final argument.
+
+Call syntax is syntactically equivalent to a primary expression followed by a
+tuple literal, except that a tuple literal requires a trailing comma to form a
+single-element tuple `(b,)`, whereas in call syntax both `a(b)` and `a(b,)` are
+permitted.
+
+### Direct calls
+
+A call expression is a _direct call_ when the callee:
+
+-   is the name of a parameterized entity, like a generic class or interface, or
+-   has a function type or bound method type.
+
+In a direct call, a call signature is available which is used to check the given
+arguments against the callee's declared implicit and explicit parameters. This
+checking proceeds as follows:
+
+-   Argument deduction is performed by comparing the declared parameter types
+    against the actual argument types and deducing values for implicit arguments
+    that make the types equal.
+-   Then, for each binding in the explicit parameter list in turn, all argument
+    values that have been deduced are substituted into the parameter.
+
+    -   If the parameter is a `template :!` binding, the argument expression is
+        converted to a constant value phase expression of the type of the
+        binding.
```

Comment:
I think missing a "template" here, and some expression phrase adjustments?

```suggestion
    -   If the parameter is a `template :!` binding, the argument expression is
        converted to a template constant expression phase of the type of the
        binding.
```

Reply:
Done.